src/mongocxx/include/mongocxx/v1/client_session.hpp

```diff
@@ -33,12 +46,342 @@ namespace v1 {
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(void) start_transaction();
+
+    ///
+    /// Commit the current transaction.
+    ///
+    /// @throws mongocxx::v1::server_error when a server-side error is encountered and a raw server error is available.
+    /// @throws mongocxx::v1::exception for all other runtime errors.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(void) commit_transaction();
+
+    ///
+    /// Abort the current transaction.
+    ///
+    /// @throws mongocxx::v1::exception when a client-side error is encountered.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(void) abort_transaction();
+
+    ///
+    /// The callback function to invoke within a standalone transaction.
+    ///
+    /// `this` is passed to the callback function and must be forwarded to operations within the callback function.
+    ///
+    /// The callback function may be invoked multiple times as part of retry attempt behavior.
+    ///
+    /// @warning The callback function **MUST** allow any @ref mongocxx::v1::server_error exceptions thrown by an
+    /// operation within the callback function to propagate out of the callback function for correct retry attempt
+    /// behavior. The callback function **MUST NOT** attempt to manually retry operations.
+    ///
+    using with_transaction_cb = std::function<void MONGOCXX_ABI_CDECL(client_session&)>;
+
+    ///
+    /// Start and commit a new standalone transaction using `fn`.
+    ///
+    /// The behavior is approximately equivalent to:
+    /// ```cpp
+    /// this->start_transaction(opts);
+    /// while (/* unspecified */) {
+    ///     try {
+    ///         fn();
+    ///         this->commit_transaction(); // Success.
+    ///     } catch (/* unspecified */) {
+    ///        if (/* unspecified */) {
+    ///            continue; // Retry attempt.
+    ///        } else {
+    ///            this->abort_transaction();
+    ///            throw; // Failure.
+    ///        }
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// Any user-defined exceptions thrown by `fn` will be caught and rethrown after the standalone transaction has been
+    /// aborted.
+    ///
+    /// @warning `fn` **MUST** allow any @ref mongocxx::v1::server_error exceptions thrown by an operation within `fn`
+    /// to propagate out of `fn` for correct retry attempt behavior. `fn` **MUST NOT** attempt to manually retry
+    /// operations.
+    ///
+    /// @note This operation uses an internal, non-configurable time limit of 120 seconds. The transaction may be
+    /// retried until this time limit is exceeded.
+    ///
+    /// @throws mongocxx::v1::exception when a client-side error is encountered.
+    /// @throws mongocxx::v1::server_error when a server-side error is encountered and all retry attempts have failed.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(void) with_transaction(with_transaction_cb const& fn, v1::transaction const& opts);
+
+    ///
+    /// Start and commit a new standalone transaction using `fn`.
```

Comment:
Suggest referring to overload with `opts` argument to avoid duplicate documentation block:

```c++
///
/// Equivalent to calling
/// @ref v1::client_session::with_transaction(with_transaction_cb const& fn, v1::transaction const& opts)
/// with default transaction options.
///
```